Automattic/wp-calypso
# Plugin Management: Refactor actions and statuses

This pull request contains only the refactoring changes that were added in #80281. See that PR for further info.

Is depended on by #80281.
Related to `1202619025189113-as-1202979359232070`.

## Proposed Changes

* Additions:
  * Create a new hook `useStatusMessageText`, to render action status messages.
  * Create a new hook `useEmptyMessage`, to render messages when no plugins are found.
* Changes:
  * Rename the `RenderStatusMessage` component to `StatusMessage`.
  * Replace usages of `getPluginActionStatusMessages` with `useStatusMessageText`.
  * Replace usages of `pluginsEmptyMessage` with `useEmptyMessage`.
  * Replace usages of `renderBulkActionsHeader` with `BulkActionsHeader`.
  * Extract all action status messages into their own files (in `plugin-action-status/messages`), grouped by the action type.
* Removals:
  * the `renderBulkActionsHeader` function from `PluginManagementV2` (in favor of the `BulkActionsHeader` component)
  * `pluginsEmptyMessage` from `utils.ts` (in favor of `useEmptyMessage`)
  * `getPluginActionStatusMessages` from `utils.ts` (in favor of `useStatusMessageText`)

## Testing Instructions

* In your test environment, with this PR applied, visit `/plugins/manage` in either Calypso Blue or Calypso Green.
* Take various actions to manage plugins on one or more of your sites. Verify that all messages appear correctly, match the action you selected, and that the status of the selected plugins matches what you expect while the actions are performed (e.g., "Activating," then "Activated"; or, "Deactivating on 4 sites"; "Deactivated on 4 sites")
* Inspect the PR's code changes, asking plenty of questions to ensure the changes all make sense.
